platform/keysystem/worldconfig/keyserver.go

```diff
@@ -0,0 +1,279 @@
+}
+
+func GenerateLocalConf(conf *SpireSetup, node *SpireNode) string {
+	scheduleWork := node.IsWorker()
+
+	return `# generated automatically by keyserver
+HOST_NODE=` + node.Hostname + `
+HOST_DNS=` + node.DNS() + `
+HOST_IP=` + node.IP + `
+SCHEDULE_WORK=` + strconv.FormatBool(scheduleWork) + `
+KIND=` + node.Kind
+}
+
+func GrantsForNodeAccount(c *config.Context, conf *SpireSetup, groups Groups, auth Authorities, ac *account.Account, node *SpireNode) map[string]account.Privilege {
+	// NOTE: at the point where this runs, not all accounts will necessarily be registered with the context!
+	var grants = map[string]account.Privilege{}
+
+	// MEMBERSHIP IN THE CLUSTER
+
+	if node.IsSupervisor() {
+		grants[BootstrapKeyserverTokenAPI] = account.NewBootstrapPrivilege(groups.Nodes, time.Hour, c.TokenVerifier.Registry)
+		grants[ImpersonateKerberosAPI] = account.NewImpersonatePrivilege(c.GetAccount, groups.KerberosAccounts)
+	}
+
+	grants[RenewKeygrantAPI] = account.NewTLSGrantPrivilege(auth.Keygranting, false, OneDay*40, ac.Principal, nil)
+
+	// CONFIGURATION ENDPOINT
+
+	grants[LocalConfAPI] = account.NewConfigurationPrivilege(GenerateLocalConf(conf, node))
+
+	// SERVER CERTIFICATES
+
+	grants[SignSSHHostKeyAPI] = account.NewSSHGrantPrivilege(
+		auth.SshHost, true, OneDay*60, "admitted-"+ac.Principal,
+		[]string{
+			node.DNS(),
+			node.Hostname,
+			node.IP,
+		},
+	)
+
+	if node.IsMaster() {
+		grants[SignKubernetesMasterAPI] = account.NewTLSGrantPrivilege(
+			auth.Kubernetes, true, 30*OneDay, "kube-master-"+node.Hostname,
+			[]string{
+				node.DNS(),
+				node.Hostname,
+				"kubernetes",
+				"kubernetes.default",
+				"kubernetes.default.svc",
+				"kubernetes.default.svc." + conf.Cluster.InternalDomain,
+				node.IP,
+				conf.Addresses.ServiceAPI,
+			},
+		)
+		grants[SignEtcdServerAPI] = account.NewTLSGrantPrivilege(
+			auth.EtcdServer, true, 30*OneDay, "etcd-server-"+node.Hostname,
+			[]string{
+				node.DNS(),
+				node.Hostname,
+				node.IP,
+			},
+		)
+	}
+
+	if node.IsSupervisor() {
+		grants[SignRegistryHostAPI] = account.NewTLSGrantPrivilege(
+			auth.ClusterCA, true, 30*OneDay, "homeworld-supervisor-"+node.Hostname,
+			[]string{"homeworld.private"},
+		)
+	}
+
+	// CLIENT CERTIFICATES
+
+	grants[SignKubernetesWorkerAPI] = account.NewTLSGrantPrivilege(
+		auth.Kubernetes, true, 30*OneDay, "kube-worker-"+node.Hostname,
+		[]string{
+			node.DNS(),
+			node.Hostname,
+			node.IP,
+		},
+	)
+
+	if node.IsMaster() {
+		grants[SignEtcdClientAPI] = account.NewTLSGrantPrivilege(auth.EtcdClient, false, 30*OneDay, "etcd-client-"+node.Hostname,
+			[]string{
+				node.DNS(),
+				node.Hostname,
+				node.IP,
+			},
+		)
+		grants[FetchServiceAccountKeyAPI] = account.NewFetchKeyPrivilege(auth.ServiceAccount)
+	}
+
+	return grants
+}
+
+func ValidateStaticFiles(context *config.Context) error {
+	for _, static := range context.StaticFiles {
+		// check for existence
+		if _, err := os.Stat(static.Filepath); err != nil {
```

Comment:
could check `IsDir()` if you feel like it

Reply:
Meh.

Reply:
i saw that lol